### PR TITLE
Expose groupsOf from Data.Stream

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/2359

File: hie.yaml [hunk 1]
```diff
@@ -94,6 +94,8 @@ cradle:
           cabal:
             - path: "./test/Streamly/Test/Data/Array.hs"
               component: "test:Data.Array"
+            - path: "./test/Streamly/Test/Data/Stream.hs"
+              component: "test:Data.Array"
```
Comment (harendra-kumar) [inline hie.yaml:98]: Should it be Data.Stream instead?

File: core/src/Streamly/Internal/Data/Stream/StreamD/Type.hs [hunk 2]
```diff
@@ -1903,6 +1903,18 @@ foldMany (Fold fstep initial extract) (Stream step state) =
...
+--
+-- >>> Stream.toList $ Stream.groupsOf 2 Fold.sum (Stream.enumerateFromTo 1 10)
+-- [3,7,11,15,19]
+--
+-- This can be considered as an n-fold version of 'take' where we apply
+-- 'take' repeatedly on the leftover stream until the stream exhausts.
+--
+-- @groupsOf n f = foldMany (FL.take n f)@
+--
+-- @since 0.9.0
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/StreamD/Type.hs:1917]: remove this line.

File: core/src/Streamly/Internal/Data/Stream/StreamD/Type.hs [hunk 3]
```diff
@@ -1903,6 +1903,18 @@ foldMany (Fold fstep initial extract) (Stream step state) =
...
+-- | Group the input stream into groups of @n@ elements each and then fold each
+-- group using the provided fold function.
+--
+-- >>> Stream.toList $ Stream.groupsOf 2 Fold.sum (Stream.enumerateFromTo 1 10)
+-- [3,7,11,15,19]
+--
+-- This can be considered as an n-fold version of 'take' where we apply
+-- 'take' repeatedly on the leftover stream until the stream exhausts.
+--
+-- @groupsOf n f = foldMany (FL.take n f)@
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/StreamD/Type.hs:1915]: Put this under doctest comments.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: core/src/Streamly/Internal/Data/Stream/StreamD/Type.hs [hunk 4]
```diff
@@ -1903,6 +1903,18 @@ foldMany (Fold fstep initial extract) (Stream step state) =
...
+--
+-- | @groupsOf n f = foldMany (FL.take n f)@
+--
+-- >>> Stream.toList $ Stream.groupsOf 2 Fold.sum (Stream.enumerateFromTo 1 10)
+-- [3,7,11,15,19]
+--
+-- This can be considered as an n-fold version of 'take' where we apply
+-- 'take' repeatedly on the leftover stream until the stream exhausts.
+--
+-- @since 0.9.0
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/StreamD/Type.hs:1917]:
```suggestion
```

File: core/src/Streamly/Internal/Data/Stream/StreamD/Type.hs [hunk 5]
```diff
@@ -1903,6 +1903,18 @@ foldMany (Fold fstep initial extract) (Stream step state) =
     step' _ (FoldManyYield b next) = return $ Yield b next
     step' _ FoldManyDone = return Stop
 
+-- | Group the input stream into groups of @n@ elements each and then fold each
+-- group using the provided fold function.
+--
+-- | @groupsOf n f = foldMany (FL.take n f)@
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/StreamD/Type.hs:1909]: Can you check this doc in haddock?

**Pull request merged**